### ISSUE TITLE
Using the UI is completely unclear to me.. 

### ISSUE THREAD

**Issue by burtonator** [hawtio/hawtio#1943]
Hawt.io seems like something that would be really awesome, and really useful to me, but I"m totally and completely confused at how to use it and it just seems completely broke.

I download the local app/jar and ran it via java -jar on port 8080.

I have a jolokia installed under ActiveMQ and I know it's working because I can curl it.

I add a new connection, try to view it, and it immediately opens a new tab.  I didn't tell it to open a new tab. That's very confusing. It shouldn't do that.  

Then... it doesn't actually show me my connection.  It just shows me help documentation. 

It's not clearly, visually, that this is the help documentation and I'm not sure why the docs should be embedded into the UI to the point where they feel like they are part of the app.

Then.. I click "do not show welcome page again"

... it's just constantly redirecting to help.

I can't view the connection.

I see now javascript errors on the console.  I was thinking it might be an XSS issue but that doesn't appear to be the case.

Long story short.  From my perspective, it's completely broken and doesn't work.. IT's frustrating to invest more time here

**Comment by davsclaus** [comment]
Kevin thanks for the feedback. 

Though instead of this "rant" you may do an effort and log tickets with more details about technical failures. Versions used, stacktraces and so on. I would have expected more from you

To use hawtio in activemq you can embed it as Dejan has blogged about
http://sensatic.net/activemq/activemq-and-hawtio.html

**Comment by burtonator** [comment]
I wan't trying to rant actually. I was trying to be constructive and make it clear that the initial experience is very confusing.  

This is why I said that I thought hawt was compelling but that it was very confusing for me to use.  

I don't really even know where to begin to break down bugs / issues because I couldn't get it working.

**Comment by davsclaus** [comment]
I suggest to either follow Dejans blog to install hawtio inside activemq, or if you run activemq on the same host as hawtio-app, then in the connect plugin, click the local tab, and "ActiveMQ" should appear. Then click that button on the right hand side, that setup the "local agent connection" and then there is a link you click and you are connected.

The remote jolokia / CORS / proxy thingy is sometimes a bit tricky to work due different browsers / CORS / activemq version changing stuff / hawtio changing proxy code to work in karaf et all. I agree that part can be better "ui experience" to report better errors.

Users of Fuse have hawtio already setup and embedded and that is what most people use it. And also users of Karaf / SMX with AMQ included have hawtio easily installed by just installing hawtio on the container and it works out of the box.

The ActiveMQ community was very unfriendly of the idea of making ActiveMQ more ready for using 3rd party web consoles. Despite there are at least 3 web consoles, where as 2 from commercial companies (not free), and then hawtio as the only 100% FOSS ASL2 licesend.

**Comment by burtonator** [comment]
Thanks.  I think the confusion with the 'rant' was I was trying to serialize what I saw on the screen.  I think I was experiencing these "UI errors".  In retrospect I should have made a screencast which would probably explain it better.  If I have the problem again I'll create a screencast.

Weird that they were unfriendly about adopting the UI.  Especially if there was no alternative.  Plan B is to always remove it in the future. 

Anyway.  Perhaps we can just embed it in our app.

I'll take a look.

thanks!

**Comment by davsclaus** [comment]
I logged a ticket about the remote connecting to ActiveMQ dont appear to work in #1944

**Comment by jstrachan** [comment]
@burtonator when folks hit issues with hawtio and some JVM with some software inside; there are unfortunately lots of things that can possibly go wrong; so giving details of the browser, OS, JVM, software versions and configurations can really help & give us a clue of how to reproduce issues. Sometimes its as simple as a browser version causing some JS exception in the JS console. (There can often be useful tell tale signs in the JS console log).

**Comment by davsclaus** [comment]
I think the connect plugin can have a easier user experience if we use wizard style for configure the options, and then have a "test" button or something. The test can maybe call the jolokia version api and see if that works?

Also if it had a known set of containers, so if you choose ActiveMQ then it knows that the path is /api/jolokia and port is 8161. 

Where as others is maybe just /jolokia and port 8080?

**Comment by burtonator** [comment]
Maybe testing step by step...  and showing what works so it can show an error on the first problem.

I didn't really have a way to debug it easily.  No errors on the console (either in the browser or java).  No UI errors, etc.

I'll try to get some time to see if I can embed it directly.

**Comment by davsclaus** [comment]
The connect plugin has been improved a bit, as it no longer shows the welcome screen again when connecting to other JMVs - #1948

**Comment by davsclaus** [comment]
I have logged a ticket to allow the connect plugin to remember credentials as it does not do this by default - https://github.com/hawtio/hawtio/issues/1947

For developers its much easier if it just remembers this. So we should likely have a [x] remember me
